include/PetriEngine/Reachability/ReachabilitySearch.h

```diff
@@ -196,6 +206,100 @@ namespace PetriEngine {
             _max_tokens = states.maxTokens();
             return false;
         }
+
+        template<typename W, typename G>
+        bool ReachabilitySearch::tryReachRandomWalk(std::vector<std::shared_ptr<PQL::Condition> >& queries,
+                                                    std::vector<ResultPrinter::Result>& results, bool usequeries,
+                                                    bool printstats, size_t maxSteps, size_t seed)
+        {
+            // set up state
+            searchstate_t ss;
+            ss.enabledTransitionsCount.resize(_net.numberOfTransitions(), 0);
+            ss.expandedStates = 0;
+            ss.exploredStates = 1;
+            ss.heurquery = queries.size() >= 2 ? std::rand() % queries.size() : 0;
+            ss.usequeries = usequeries;
+
+            // set up working area
+            Structures::State state;
+            Structures::State working;
+            _initial.setMarking(_net.makeInitialMarking());
+            state.setMarking(_net.makeInitialMarking());
+            working.setMarking(_net.makeInitialMarking());
+
+            W states(_net, _kbound);                        // stateset
```

Comment:
This "stateset" should not exist in the random walk. (or rather, if/when merged with the normal reachability, it should do nothing)